sound-equipment-fault-detection/webui/controllers/baseController.go

```diff
@@ -0,0 +1,135 @@
+package controllers
+
+import (
+	"context"
+	"flag"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+// var kubeconfig *string
+
+// func init() {
+// 	// kubeconfig = flag.String("kubeconfig", "./config", "(optional) absolute path to the kubeconfig file")
+// }
+
+// This is a placeholder for your LED status
+var ledStatus = true
+
+// Handler function to get the LED status
+func GetLEDStatus(c *gin.Context) {
+	value := GetValue()
+
+	if value == "0" {
+		ledStatus = false
+	} else if value == "1" {
+		ledStatus = true
+	} else {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "Invalid value"})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"isOn": ledStatus,
+	})
+}
+
+func GetValue() string {
+	flag.Parse()
+
+	var config *rest.Config
+	var err error
+
+	// If a kubeconfig file is specified, use external config
+	// if *kubeconfig != "" {
+	// 	config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+
+	// } else {
+	// 	// Otherwise use the cluster internal configuration
+	// 	config, err = rest.InClusterConfig()
+	// }
+
+	config, err = rest.InClusterConfig()
```

Comment:
Use the InClusterConfig() function to get the Kubernetes cluster configuration instead of relying on the kubeconfig file.